lib/OpenQA/CacheService/Model/Cache.pm

```diff
@@ -69,6 +69,7 @@ sub _download_asset {
     my $ua  = $self->ua;
     my $url = sprintf '%s/tests/%d/asset/%s/%s', $host, $id, $type, basename($asset);
     $log->info('Downloading "' . basename($asset) . qq{" from "$url"});
+    $url =~ s/#/%23/g;
```

Comment:
I was more thinking of generating a valid URL in the first place by not using "sprintf" but a method that properly concatenates string parts into a valid URL.

@kraih maybe you have an immediate suggestion? Do it with a new URL object using https://mojolicious.org/perldoc/Mojo/URL ?

Reply:
Yes, better use Mojo::URL for making the URL.

Reply:
Indeed, this should build the URL properly with `Mojo::URL`.